internal/eventstore/handler/handler_projection.go

```diff
@@ -125,18 +126,26 @@ func (h *ProjectionHandler) Trigger(ctx context.Context, instances ...string) er
 // Process handles multiple events by reducing them to statements and updating the projection
 func (h *ProjectionHandler) Process(ctx context.Context, events ...eventstore.Event) (index int, err error) {
 	if len(events) == 0 {
-		return 0, nil
+		return index, nil
 	}
 	index = -1
-	statements := make([]*Statement, len(events))
-	for i, event := range events {
-		statements[i], err = h.reduce(event)
-		if err != nil {
-			return index, err
+	statements := make([]*Statement, 0)
+	for _, event := range events {
+		statement, reduceErr := h.reduce(event)
+		if reduceErr != nil {
+			return index, reduceErr
+		}
+		// statement can be nil (see notifications)
+		if statement != nil {
+			statements = append(statements, statement)
 		}
 	}
+	// update panics if there are no statements
+	if len(statements) == 0 {
+		return index, caos_errors.ThrowInternal(nil, "HANDL-vjxj7", "Errors.Handlers.Process.NoStatements")
+	}
 	for retry := 0; retry <= h.retries; retry++ {
-		index, err = h.update(ctx, statements[index+1:], h.reduce)
+		index, err = h.update(ctx, statements, h.reduce)
```

Comment:
Don't you change the logic here if the returned value gets ignored in the retry?

Reply:
Would also prefer if there can be added until tests for this situation

Reply:
discarded these changes and added unit tests for configuration option